### DISCUSSION TITLE
disable Deno object in javascript

### DISCUSSION THREAD

**Discussion by amirhosseinvz94** [denoland/deno#13722] [Q&A]
Hi,

I'm executing untrusted javascript (users script) inside deno, and i need to disable users to use Deno functionality like Deno.core or Deno.core.opAsync or anything related to Deno, how to do it?

**Comment by kitsonk** [comment]
You simply need to delete `Deno.core` before you run any of the untrusted code.  For example:

**delete_core.js**

```js
delete Deno.core;
```

**main.js**

```js
import "./delete_core.js";

console.log(Deno.core); // undefined
```

All internal code binds early to `Deno` and `Deno.core` so deleting them won't impact downstream runtime code.

**Comment by amirhosseinvz94** [comment]
thanks mr kitsonk
but i need this `Deno.`  api In the middle of executing the user code for myself(my handler functions ) or after running user script, my user do not have access to any concept of pure deno ( like any api for example `Deno.core` or `Deno.Error` or `...` or any returned `error` from deno or anything ), somehow i do not want to user understand that behind of engine is deno
i think we can handle it in javascript with wrapper ( wrap Deno object with null ), but i think absolutely it has hole, And the user can get away with it, and i think maybe exist one way in rust side to user limitation and i think it is better way because user access to rust side is difficult

**Reply by zandaqo** [reply]
```js
export const internalDeno = Deno.core;
delete Deno.core;
````
```js
import { internalDeno } from 'delete_core.js';
console.log(Deno.core); // undefined
console.log(internalDeno); // { ... }
```

**Comment by leoc11** [comment]
Maybe you should consider running it on worker. By default the Deno namespace is not available in worker scope.

**Reply by amirhosseinvz94** [reply]
You mean web work?
